### PR TITLE
[CAPI] Template types for `MachineDeployment`s etc.

### LINKED ISSUE #166: We need `K0sWorkerConfigTemplate` type
https://cluster-api.sigs.k8s.io/developer/providers/bootstrap.html#bootstraptemplate-resources

This is needed to be able to use `MachinePool` etc. resources in cluster api.

### PR DESCRIPTION
**Pull request by jnummelin**
Fixes #166

### CODE REVIEW

**Comment by jnummelin** [comment]
@makhov @juanluisvaladas PTAL

File: inttest/capi-docker-machinedeployment/capi_docker_test.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+			return false, nil
+		}
+		return true, nil
+	})
+	s.Require().NoError(err)
+}
+
+func (s *CAPIDockerSuite) applyClusterObjects() {
+	// Exec via kubectl
+	out, err := exec.Command("kubectl", "apply", "-f", "../../config/samples/capi/docker/cluster-with-machinedeployment.yaml").CombinedOutput()
```
[the remark this answers is not in the document]
Comment (jnummelin) [reply inttest/capi-docker-machinedeployment/capi_docker_test.go:135]: yeah, bit sketchy I agree. But I've had some troubles creating the cluster API object via dynamic client. We do need to figure that part out but maybe that can be separate task/PR?

File: docs/capi-bootstrap.md [hunk 2]
```diff
@@ -39,3 +39,61 @@ As k0s comes with all the needed bells and whistles to get k8s worker node up, w
 Check the [examples](capi-examples.md) pages for more detailed examples how k0smotron can be used with various Cluster API infrastructure providers.
 
 For reference on what can be configured via `K0sWorkerConfig` see the [reference docs](resource-reference.md#bootstrapclusterx-k8siov1beta1).
+
+## MachineDeployments, ..Sets etc.
```
[the remark this answers is not in the document]
Comment (jnummelin) [reply docs/capi-bootstrap.md:43]: Yes, makes more sense

File: inttest/capi-docker-machinedeployment/capi_docker_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (juanluisvaladas) [reply inttest/capi-docker-machinedeployment/capi_docker_test.go:135]: Well it's a test so it's not the end of the world. When we generate the client like in k0s we can change it...

**Review by juanluisvaladas — APPROVED** [review]

**Pull request merged**